File: CryptoCurrenciesApp/CryptoCurrenciesApp/Domain/Currency/CryptoCurrencyRepository.swift
//
//  CryptoCurrencyRepository.swift
//  CryptoCurrenciesApp
//
//  Created by Алексей Рамашка on 28.01.21.
//

import Foundation
import RxSwift

protocol CryptoCurrencyRepository {
    func getCryptoCurrenciesList(countOfItems: Int) -> Single<Array<CryptoCurrencyModel>>
}

File: SDK/sdk/domain/src/commonMain/kotlin/by.st.kmm.sdk.domain/currency/CryptoCurrencyRepository.kt
package by.st.kmm.sdk.domain.currency

import com.badoo.reaktive.observable.Observable
import com.badoo.reaktive.single.Single

/**
 * @author RamashkaAE
 */
interface CryptoCurrencyRepository {

    fun getCryptoCurrenciesList(countOfItems: Int) : Single<List<CryptoCurrencyModel>>

    fun startInitialization(countOfItems: Int) : Observable<Int>
}

class RepositoryException(val errorCode: Int, val errorMessage: String?) : Exception(errorMessage)